src/library/export/coverartcopyworker.cpp

```diff
@@ -0,0 +1,112 @@
+#include "library/export/coverartcopyworker.h"
+
+#include <QDebug>
+#include <QFileInfo>
+#include <QMessageBox>
+
+#include "util/imagefiledata.h"
+#include "util/safelywritablefile.h"
+
+void CoverArtCopyWorker::run() {
+    m_isCoverArtUpdated = false;
+    copyFile(m_coverArtImage, m_coverArtAbsolutePath);
+}
+
+void CoverArtCopyWorker::copyFile(
+        const ImageFileData& m_coverArtImage,
+        const QString& m_coverArtAbsolutePath) {
+    QFileInfo coverArtPathFileInfo(m_coverArtAbsolutePath);
+
+    if (coverArtPathFileInfo.exists()) {
+        switch (makeOverwriteRequest(m_coverArtAbsolutePath)) {
+        case OverwriteAnswer::CANCEL:
+            return;
+        case OverwriteAnswer::OVERWRITE:
+            break;
+        }
+
+        mixxx::SafelyWritableFile safelyWritableFile(m_coverArtAbsolutePath, true);
```

Comment:
This is not used. We have the issue here, that it adds the extension "temp" and than requires to write to that file . 
This will fail our assertions in ImageFileData ...